provider/maas/environ.go

```diff
@@ -103,15 +102,27 @@ type maasEnviron struct {
 
 	// apiVersion tells us if we are using the MAAS 1.0 or 2.0 api.
 	apiVersion string
+
+	// GetCapabilities is a function that connects to MAAS to return its set of
+	// capabilities.
+	GetCapabilities MaasCapabilities
 }
 
 var _ environs.Environ = (*maasEnviron)(nil)
 
-func NewEnviron(cloud environs.CloudSpec, cfg *config.Config) (*maasEnviron, error) {
+// MaasCapabilities is just a shorthand for this very long function definition.
+// It represents a function that gets the capabilities of a MAAS installation.
+type MaasCapabilities func(client *gomaasapi.MAASObject, serverURL string) (set.Strings, error)
+
+func NewEnviron(cloud environs.CloudSpec, cfg *config.Config, getCaps MaasCapabilities) (*maasEnviron, error) {
+	if getCaps == nil {
+		getCaps = getCapabilities
+	}
```

Comment:
Well, the entire point is to test `getCapabilities` separately from the type(s) that use it, so exposing it falls along those lines.